src/org/usfirst/frc/team1923/robot/subsystems/VisionSubsystem.java

```diff
@@ -18,85 +27,133 @@
  */
 public class VisionSubsystem extends Subsystem {
 
-	public NetworkTable table;
-	public double[] x;
 	public double centerx,turn;
 	public double[] widtharr;
 	public double width;
 	public double dist;
 	
 	public boolean found;
 	
-	private double sum;
+	private double sumx;
+	private double sumw;
 	private double[] def; 
 	
+	public double centerX;
+	
 	public Ultrasonic frontSonar;
 	
+	 Mat source = new Mat();
+     Mat output = new Mat();
+     CvSink cvSink;
+     CvSource outputStream;
+     GripPipeline pipe;
+     Rect r;
+	
 	/**
 	 * Initializes CameraServer and NetworkTables
 	 */
 	public VisionSubsystem() {
+		try{
 		//TODO: Implement Bounding Rectangle
 		//TODO: IMplement Controller Vibration when Match time is getting low
 		//Start Camera Server
-		CameraServer.getInstance().addAxisCamera(RobotMap.CAMERA_IP);
+		//CameraServer.getInstance().addAxisCamera(RobotMap.CAMERA_IP);
 		//Testing Drawing Bounding Rectangle Around Peg
 		/*
 		CvSource output = CameraServer.getInstance().putVideo("Annotated", 320, 240);
 		*/
 		//TODO: Account for difference in areas of tape to change turn value
 		//TODO: Add ultrasonic sensors
 		def = new double[0];
-		table = NetworkTable.getTable(RobotMap.NEWTORK_TABLE_ADDRESS);
 		frontSonar = new Ultrasonic(RobotMap.FRONT_SONAR_PING_PORT, RobotMap.FRONT_SONAR_ECHO_PORT, Unit.kInches);
 		frontSonar.setEnabled(true);
 		frontSonar.setAutomaticMode(true);
 		dist=frontSonar.getRangeInches();
 		found=false;
+
+		AxisCamera camera = CameraServer.getInstance().addAxisCamera("Axis Cam", RobotMap.CAMERA_IP);
+		camera.setResolution(320, 240);
+        
+        cvSink = CameraServer.getInstance().getVideo();
+        outputStream = CameraServer.getInstance().putVideo("Processed", 320, 240);
+        
+        SmartDashboard.putNumber("Power", 0);
+        
+        pipe = new GripPipeline();
+       	
 		refresh();
+		}
+		catch(Exception e){
+			System.out.println("Exception was thrown: " + e);
+		}
 	}
 	
 	public void refresh(){
-		dist=frontSonar.getRangeInches();
-		x = table.getNumberArray("centerX", def);
-		widtharr= table.getNumberArray("width", def);
-		sum=0;
-		for(double a: widtharr){
-			sum+=a;
+		//TODO: Move refresh method to a seperate command to get automatic multithreading
+		try{
+		//Process Image
+		try{

```

Comment:
A single `try-catch` block could be used here instead of nested `try-catch` blocks